doc/faq.rst

```diff
@@ -87,3 +89,35 @@ You can build pyopencl against conda's OpenCL driver in the following way::
    # Apply this patch on MacOS: https://raw.githubusercontent.com/conda-forge/pyopencl-feedstock/master/recipe/osx_flags.diff
    $ ./configure.py --cl-inc-dir=$PWD/../miniforge3/envs/ceesd/include --cl-lib-dir=$PWD/../miniforge3/envs/ceesd/lib
    $ pip install -e .
+
+How can I record the exact versions of Python packages that are currently installed and reinstall them at a later time?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Running emirge's `version.sh` script creates a requirements.txt file that
+stores the exact git commits of each emirge sub-repository that were used at
+the time when `version.sh` was executed. You can use this file to install the
+exact versions of the packages at a later time::
+
+   $ cd emirge/
+   $ ./version.sh --output-requirements=myreq.txt
+   [...]
+   *** Creating requirements file with current emirge module versions
+   [...]
+   *** Created file 'myreq.txt'. Install it with 'pip install --src . -r myreq.txt'.
+
+   $ pip install --src . -r myreq.txt
+
+
+.. note::
+
+   This will build pyopencl by source, which can be challenging on some systems. Please
+   see :ref:`Pyopencl source installation` for information on the prerequisites of Pyopencl,
+   or remove/comment the pyopencl line from the generated requirements file.
+
+.. note::
+
+   You can also install the packages from the created requirements.txt into a new conda environment::
+   
+   $ conda create -n new_env
+   $ conda activate new_env
+   $ pip install --src . -r myreq.txt
```

Comment:
I figured that the `requirements.txt` could be provided to `install.sh` via a flag?

Reply:
No, that won't work for now. Our parser doesn't handle the `--editable` flag inside the requirements.txt.

Reply:
It should. Otherwise why are we building systems that don't make sense and where nothing fits together?

Reply:
Done in https://github.com/illinois-ceesd/emirge/pull/88